SubscribeIT/ngDesk
# 287 sidebar improvement

* Added a json file called sidebar,json
* Modified postDefaultSidebar function in order to post side bar menu items based on the plugins selected by the user(only for 
* system admin role, for other roles system admin can add).
* Tickets and pager will be posted by default.
* If user selects chats it will be added for both system admin and agents by default.
* Changed plugin name "Software Asset Management" to "SAM"
* Removed old and unwanted sidebar files
* Everything is working fine.

Test Cases:-
Test Case 1:-
Steps:
* Post a new a company.
* while posting add plugins randomly to the payload.
* login to ui and check whether menu items for the given plugins are added or not.

Test Case 2:-
Steps:
* After login to ui, try to add another menu item to the sidebar from the ui
* make sure all the crud operations works correctly

Test Case 3:-
* After login to ui, try to interchange the order of the menu items in the ui
* Make sure it works correctly

Test Case 4:-
Steps:
* when chats is added to the plugin, it should add chats for the side bar of both system admin and agents.
* When expenses is added to the plugin then only it should add the side bar for accountant,spender and accounting manager 
  roles otherwise it should not.

Test Case 5:-
Steps:
* For Agent(except for chat plugin),Sales and Customer roles, there should be only default side bar menu items.


Screenshots:-

![Screenshot from 2021-11-17 13-23-14](https://user-images.githubusercontent.com/89504492/142173812-bffe0174-e59e-405d-87af-2556cdee4cd2.png)
![Screenshot from 2021-11-17 13-28-21](https://user-images.githubusercontent.com/89504492/142173826-e1e22861-23d4-4140-8e60-5e53bd45fd50.png)
![Screenshot from 2021-11-17 13-29-18](https://user-images.githubusercontent.com/89504492/142173837-12f3ed9e-be8d-4dce-89ab-f564c0dd8fb6.png)
![Screenshot from 2021-11-17 13-29-25](https://user-images.githubusercontent.com/89504492/142173844-e1b4b70a-3f51-497d-8fe9-6f094e6ce364.png)
